pkg/tfshim/sdk-v2/provider2.go

```diff
@@ -147,6 +151,73 @@ type planResourceChangeImpl struct {
 
 var _ planResourceChangeProvider = (*planResourceChangeImpl)(nil)
 
+// If the user specifies custom timeout overrides for the resource, encode them in the magic `timeouts` key under config
+// and plannedState. This is how TF communicates this information on the gRPC protocol. The schema default timeouts need
+// not be specially encoded because the gRPC implementation already respects them. A slight complication here is that
+// some resources do not seem to allow customizing timeouts for certain operations, and their impliedType will not have
+// the corresponding slot. Warn the user if the custom timeout is a no-op.
+func (p *planResourceChangeImpl) configWithTimeouts(
+	ctx context.Context,
+	configWithoutTimeouts map[string]any,
+	topts shim.TimeoutOptions,
+	impliedType cty.Type,
+) map[string]any {
+	config := map[string]any{}
+	for k, v := range configWithoutTimeouts {
```

Comment:
What is `configWithoutTimeouts`? Is it the TF config translation of the Pulumi config? A comment explaining that would be helpful.

It would be amazing if `configWithoutTimeouts` was it's own type, but that's a painful pattern in Go.

Reply:
Yeah. I can use ResourceConfig which is what's needed, but I always forget how to construct that. All the helper types aren't quite there yet to be truly usable. Added some commits. 

Generally I agree with your sentiment, just qualifying it with this idea that I sweat the signatures a bit less for functions that are private file-local single-use functions, e.g. the smaller the scope the less I worry about this. But I'm quite open to efforts to raise the bar on code quality in this codebase so if you have plans here happy to play along or follow a style guide.